bdns/dns.go

```diff
@@ -303,3 +291,68 @@ func (dnsResolver *DNSResolverImpl) LookupMX(hostname string) ([]string, error)
 
 	return results, nil
 }
+
+// exchangeOne performs a single DNS exchange with a randomly chosen server
+// out of the server list, returning the response, time, and error (if any).
+// This method sets the DNSSEC OK bit on the message to true before sending
+// it to the resolver in case validation isn't the resolvers default behaviour.
+func (dnsResolver *DNSResolverImpl) exchangeOne(ctx context.Context, hostname string, qtype uint16, msgStats metrics.Scope) (*dns.Msg, error) {
```

Comment:
Why move this function? It makes it harder to review the diffs, and complicates future `git blame`. I'm happy to move functions around if there's a reason (e.g. splitting into multiple files), but I don't see the reason here.